src/usr/share/netcheck/networkmanagerhelper.py

```diff
@@ -413,6 +420,28 @@ def connection_is_activated(self, connection_id):
 
         return connection_is_activated
 
+    def get_connection_for_interface(self, interface_name):
+        """Find and return the connection id currently applied to the given interface.
+
+        interface_name: The system interface name, for example eth1 or ens0s1.
+        """
+        connection_id = None
+        devices = []
+
+        try:
+            devices = self.NetworkManager.NetworkManager.GetDevices()
+        except Exception as exception:
+            self.logger.error(
+                'Unable to get device list from NetworkManager. %s: %s.',
+                type(exception).__name__, str(exception))
+
+        for device in devices:
+            if device.Interface == interface_name:
+                connection_settings = self._get_applied_connection(device)
+                connection_id = connection_settings['connection']['id']
```

Comment:
If there is no applied connection, this will throw an error. Even if the calling code has established that there is a connection, this code is racy and the connection might not exist by the time the code reaches this point. You should check to make sure connection_settings is not None.